### PR TITLE
Modify device_id setting way to avoid ambiguity while setting device_id by env variable or yaml

### PR DESCRIPTION
**Pull request by HaoyangLee**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:
- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

## Motivation
There are two ways to specify device id while standalone training:
- (1) export DEVICE_ID=7
- (2) set 'device_id=7' in system section of yaml config file

We take (1) as the higher priority, namely, (2) is only valid when distribute=False (standalone training) and environment variable 'DEVICE_ID' is NOT set.

### CODE REVIEW

File: configs/cls/mobilenetv3/cls_mv3.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 system:
   mode: 0 # 0 for graph mode, 1 for pynative mode in MindSpore
   distribute: True
-  device_id: 7  # only valid when distribute=True, i.e., standalone training
+  device_id: 7  # only valid when distribute=True (standalone training) and environment variable 'DEVICE_ID' is NOT set
```
Comment (hadipash) [inline configs/cls/mobilenetv3/cls_mv3.yaml:4]: Did you mean False?

Comment (hadipash) [inline configs/cls/mobilenetv3/cls_mv3.yaml:4]: When distribute is False.

File: docs/en/tutorials/yaml_configuration.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ This document takes `configs/rec/crnn/crnn_icdar15.yaml` as an example to descri
 | ---- | ---- | ---- | ---- | ---- |
 | mode | Mindspore running mode (static graph/dynamic graph) | 0 | 0 / 1 | 0: means running in GRAPH_MODE mode; 1: PYNATIVE_MODE mode |
 | distribute | Whether to enable parallel training | True | True / False | \ |
+| device_id | Specify the device id while standalone training | 7 | The ids of all devices in the server | Only valid when distribute=True (standalone training) and environment variable 'DEVICE_ID' is NOT set. |
```
Comment (hadipash) [inline docs/en/tutorials/yaml_configuration.md:25]: distribute is False

File: configs/cls/mobilenetv3/cls_mv3.yaml (hunk 1 above)
Comment (HaoyangLee) [reply configs/cls/mobilenetv3/cls_mv3.yaml]: Fixed, thx.

File: docs/en/tutorials/yaml_configuration.md (hunk 2 above)
Comment (HaoyangLee) [reply docs/en/tutorials/yaml_configuration.md]: Fix, thx

File: docs/cn/tutorials/yaml_configuration.md
Comment (hqkate) [inline docs/cn/tutorials/yaml_configuration.md:1]: default value is 0?

Comment (hadipash) [reply docs/cn/tutorials/yaml_configuration.md:1]: It is better to keep it 7 to leave card 0 for distributed training.

Comment (HaoyangLee) [reply docs/cn/tutorials/yaml_configuration.md:1]: The default value of `device_id` in yaml is suggested to be 7. If `device_id` is not specified in yaml, mindspore will use device 0 by default.

Comment (hqkate) [reply docs/cn/tutorials/yaml_configuration.md:1]: I see, but in the table, the column represents the default number.

**Review by hadipash — APPROVED** [review]

File: docs/cn/tutorials/yaml_configuration.md
Comment (zhtmike) [reply docs/cn/tutorials/yaml_configuration.md:1]: The default value of device_id in yaml is 7 is bit strange. what if the user does not have 8 devices?

Comment (HaoyangLee) [reply docs/cn/tutorials/yaml_configuration.md:1]: > I see, but in the table, the column represents the default number.

If we have 'device_id' in yaml, the default value is 7, refer to [`configs/cls/mobilenetv3/cls_mv3.yaml`](https://github.com/mindspore-lab/mindocr/blob/main/configs/cls/mobilenetv3/cls_mv3.yaml#L4).
I add more details in yaml_configuration.md to be more clear.

Comment (HaoyangLee) [reply docs/cn/tutorials/yaml_configuration.md:1]: > The default value of device_id in yaml is 7 is bit strange. what if the user does not have 8 devices?

As Rustam said, due to the inconvenience of setting different cards for distributed training on Ascend, the default device_id (for standalone training) in yaml is supposed not to be 0. In the meantime, we don't know how many devices users have. So the compromise is to specify 'device_id=7' in yaml. If no 8 devices, let the error raise.

**Review by zhtmike — APPROVED** [review]

**Pull request merged**